MultiLangProgGuide/MultiLangProgGuide.rst

```diff
@@ -1819,6 +1821,1220 @@ If the requested :code:`timeout_nsec` exceeds
+as a parameter.
+
+Since Go does not permit a pointer to a Go function to be passed as a
+parameter to a C function, :code:`handler` is required to be a pointer
+to a C function. For a pure Go application, the C function is a glue
+routine that in turn calls the Go function. YottaDB provides a shell
+script `GenYDBGlueRoutine.sh`_ to generate glue routine functions. The
+:code:`handlerData` structure should be in memory allocated with `Go
+Malloc()`_ to protect it from Go garbage collection.
+
+Owing the need to use :code:`unsafe pointer` parameters, please take
+additional care in designing and coding your application to ensure the
+validity of the pointers passed to :code:`TimerStart()`.
+
+Go Programming Notes
+====================
+
+These `Go Programming Notes`_ supplement rather than supplant the more
+general `Programming Notes`_ for C.
+
+----------
+Goroutines
+----------
+
+As the YottaDB runtime system is not thread-safe, and because the
+internal data structures and operating system interfaces of Go are not
+part of a stable API, out of an abundance of caution, the initial
+implementation of the YottaDB Go Wrapper is restricted to executing
+its logic in a single goroutine.
+
+In order to avoid restricting Go applications from calling the YottaDB
+API from a single goroutine (which would be unnatural to a Go
+programmer), the YottaDB Go wrapper includes logic that coerces the
+execution of the YottaDB runtime system to a single
+goroutine. Directly calling YottaDB C API functions bypasses this
+protection, and may result in unpredictable results (Murphy says that
+unpredictable results will occur when you least expect
+them). Therefore, Go application code should only call the YottaDB API
+exposed in this `Programming in Go`_ section.
+
+--------------------
+GenYDBGlueRoutine.sh
+--------------------
+
+As discussed in `Go TpS()`_ and `Go TimerStart()`_, as Go does not
+permit a pointer to a Go function to be passed as a parameter to a C
+function, Go functions that encapsulate logic to be executed as an
+ACID transaction, and Go functions that serve as handlers for 
+expired timers cannot be passed as parameters to :code:`TpS()` and
+:code:`yottadb.TimerStart()`. Instead, each Go function must have a C
+“glue” routine whose address is passed to :code:`TpS()` or
+:code:`yottadb.TimerStart()`.
+
+If an application has a callback routine called :code:`CallBackRtn()`,
+executing :code:`GenYDBGlueRoutine.sh CallBackRtn` will generate a
+routine :code:`CallBackRtn_cgo.go`. In the following example,
+:code:`/usr/lib/yottadb/r122` is the directory where YottaDB r1.22
+(the YottaDB release to be used) resides.
+
+.. code-block:: go
+
+    // #cgo CFLAGS: -I/usr/lib/yottadb/r122
+    // #include <stdlib.h>
+    // #include <stdio.h>
+    // #include <errno.h>
+    // #include <string.h>
+    // #include "libyottadb.h"
+    // #cgo LDFLAGS: -L/usr/lib/yottadb/r122 -lyottadb -Wl,-rpath,/usr/lib/yottadb/r122
```

Comment:
Probably should remove all the includes in this group except libyottadb.h as they have no relevance to what needs to be added. That was just what I had in my main routine for the POC. I removed them all and it still built.

Reply:
Removed